src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -408,6 +409,134 @@ public void WriteValue(byte[] value)
             JsonValueUtils.WriteValue(this.writer, value, this.wrappedBuffer, this.ArrayPool);
         }
 
+#if NETCOREAPP3_1_OR_GREATER
+        public void WriteValue(JsonElement value)
+        {
+            switch (value.ValueKind)
+            {
+                case JsonValueKind.Null:
+                    this.WriteValue((string)null);
+                    break;
+                case JsonValueKind.False:
+                    this.WriteValue(false);
+                    break;
+                case JsonValueKind.True:
+                    this.WriteValue(true);
+                    break;
+                case JsonValueKind.String:
+                    this.WriteValue(value.GetString());
+                    break;
+                case JsonValueKind.Array:
+                    this.WriteJsonElementArray(value);
+                    break;
+                case JsonValueKind.Object:
+                    this.WriteJsonElementObject(value);
+                    break;
+                case JsonValueKind.Number:
+                    this.WriteJsonElementNumber(value);
+                    break;

```

Comment:
don't have a default? How about 'Undefined'?

Reply:
We have enumerated all known JSON types, `JsonValueKind` does actually have a `JsonValueKind.Undefined` variant which means "no value". But it's not clear to me what its use case is when parsing JSON because anything that does not map to the known JSON types (null, bool, number, string, array and object) would have resulted in an exception during parsing.

So in this case I think we have handled all known cases. I should add a `default` case, but I don't think it should write anything to the output. It should either:
- do nothing and continue (and treat this edge case as undefined behaviour), or
- throw an exception

What do you think @xuzhg ?